draft-ietf-rats-corim.md

```diff
@@ -1119,6 +1119,67 @@ An Endorsed Values triple declares additional measurements to add to the ACS.
 {::include cddl/endorsed-triple-record.cddl}
 ~~~
 
+#### Conditional Endorsement Triple {#sec-comid-triple-cond-endors}
+
+The semantics of the Conditional Endorsement Triple is as follows:
+
+> "IF accepted state matches all `conds` values, THEN every entry in the `endorsements` is added to the accepted state"
+
+~~~ cddl
+{::include cddl/conditional-endorsement-triple-record.cddl}
+~~~
+
+A `conditional-endorsement-triple-record` has the following parameters:
+
+* `conditions`: all target environments, along with a specific state, that need to match `state-triples` entries in the ACS for the endorsement(s) to apply
+* `endorsements`: endorsements that are added to the ACS `state-triples` if all `conds` match.
+
+The order in which Conditional Endorsement triples are evaluated is important: different sorting may produce different end-results in the computed ACS.
+
+Therefore, the set of applicable Conditional Endorsement triples MUST be topologically sorted based on the criterion that a Conditional Endorsement triple is evaluated before another if its Target Environment and Endorsement pair is found in any of the stateful environments of the subsequent triple.
```

Comment:
I don't think topological sorting is going to be a complete solution as profiles get more and more complex. We should be computing a fixed point. The ordering of conditions may not be easily decided, but if it is, then we can suggest a topological sort to improve convergence speed.

Reply:
See revised text.